tgui/packages/tgui/interfaces/BluespaceArtillery.js

```diff
@@ -1,64 +1,83 @@
-                    fluid
-                    content="FIRE"
-                    color="bad"
-                    disabled={!target}
-                    fontSize="30px"
-                    textAlign="center"
-                    lineHeight="46px"
-                    onClick={() => act('fire')}
-                  />
+          <Stack>
+            <Stack.Item grow={1}>
+              <Section title="Charge">
+                <ProgressBar
+                  animated
+                  step={1}
+                  minValue={0}
+                  maxValue={max_charge}
+                  value={charge}
+                  ranges={{
+                    good: [max_charge, Infinity],
+                    average: [max_charge * 0.2, max_charge * 0.99],
+                    bad: [-Infinity, max_charge * 0.2],
+                  }}>
+                  {formatted_charge}
+                </ProgressBar>
+              </Section>
+              <Section title="Target">
+                <Box color={target ? 'average' : 'bad'} fontSize="25px">
+                  {target ? target[1] : 'No Target Set'}
                 </Box>
-              ) : (
-                <>
-                  <Box color="bad" fontSize="18px">
-                    Bluespace artillery is currently locked.
+              </Section>
+              <Section>
+                {unlocked ? (
+                  <Box style={{ margin: 'auto' }}>
+                    <Button
+                      fluid
+                      content="FIRE"
+                      color="bad"
+                      disabled={!target}
+                      fontSize="30px"
+                      textAlign="center"
+                      lineHeight="46px"
+                      onClick={() => target && act('fire')}
+                    />
                   </Box>
-                  <Box mt={1}>Awaiting authorization via keycard reader from at minimum two station heads.</Box>
-                </>
-              )}
-            </Section>
-          </>
+                ) : (
+                  <>
+                    <Box color="bad" fontSize="18px">
+                      Bluespace artillery is currently locked.
+                    </Box>
+                    <Box mt={1}>Awaiting authorization via keycard reader from at minimum two station heads.</Box>
+                  </>
+                )}
+              </Section>
+            </Stack.Item>
+            <Stack.Item width="200px">
+              <Section title="Available Targets" height="100%" scrollable fill>
+                <Box height="100%">
+                  {Object.entries(targets || {}).map(([key, value], index) => (
+                    <Box key={`${key}-${index}`} mb={1}>
+                      <Button
+                        fluid
+                        color={target && key === target[0] ? 'bad' : 'blue'}
+                        content={value}
+                        onClick={() => act('set_target', { target: key })}
```

Comment:
Don't forget to change `target`

Also, for
```js
onClick={() => act('set_target', { target: key })}`
```
`target` should be `chosen_target`

Reply:
done